plugins/scorecards/src/api/ScorecardsClient.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ScorecardsApi } from './ScorecardsApi';
+import {
+  CheckResult,
+  CheckResponse,
+} from '@backstage/plugin-tech-insights-common';
+import { DiscoveryApi } from '@backstage/core-plugin-api';
+
+export type Options = {
+  discoveryApi: DiscoveryApi;
+  proxyPath?: string;
+};
+
+export class ScorecardsClient implements ScorecardsApi {
+  private readonly discoveryApi: DiscoveryApi;
+  private baseUrl: string = '';
+
+  constructor(options: Options) {
+    this.discoveryApi = options.discoveryApi;
+  }
+
+  getBaseUrl: () => Promise<string> = async () => {
+    if (!this.baseUrl) {
+      this.baseUrl = await this.discoveryApi.getBaseUrl('tech-insights');
+    }
+    return this.baseUrl;
+  };
+
+  async fetchAllChecks(): Promise<CheckResult[]> {
+    const url = await this.getBaseUrl();
+    const allChecks = await fetch(`${url}/checks`);
+    const payload = await allChecks.json();
+    if (!allChecks.ok) {
+      throw new Error(payload.errors[0]);
+    }
+    const checks = payload.map((check: CheckResponse) => check.id);
```

Comment:
This isn't really returning the `CheckResult` type right? It's only returning check ids. I think it's better to rename this method and change the return type to match expected or to create a frontend type for the actual (generic) shape of the returned object. 

This endpoint returns `TechInsightCheck` which is in the `tech-insights-node` package at the moment. If you want to use that directly (which is fine) you need to move it to `tech-insights-common` package at the same time (and make `tech-insights-node` to pipe that through as well.

Reply:
Yeah, I agree it is a bit misleading, will change that